From 7299fc2156d5574df469561db900fa711fda059a Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:24 -0400
Subject: [usb] ehci-exynos: Return immediately from suspend if ehci_suspend fails

Message-id: <1410887463-20674-373-git-send-email-dzickus@redhat.com>
Patchwork-id: 95328
O-Subject: [RHEL7 PATCH 372/591] usb: ehci-exynos: Return immediately from suspend if ehci_suspend fails
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d72175103f25783b0504f864a4f381621a789ca2
Author: Vivek Gautam <gautam.vivek@samsung.com>
Date:   Thu Apr 10 15:58:01 2014 +0530

    usb: ehci-exynos: Return immediately from suspend if ehci_suspend fails

    Patch 'b8efdaf USB: EHCI: add check for wakeup/suspend race'
    adds a check for possible race between suspend and wakeup interrupt,
    and thereby it returns -EBUSY as error code if there's a wakeup
    interrupt.
    So the platform host controller should not proceed further with
    its suspend callback, rather should return immediately to avoid
    powering down the essential things, like phy.

    Signed-off-by: Vivek Gautam <gautam.vivek@samsung.com>
    Acked-by: Jingoo Han <jg1.han@samsung.com>
    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ehci-s5p.c b/drivers/usb/host/ehci-s5p.c
index 379037f..c3f02d6 100644
--- a/drivers/usb/host/ehci-s5p.c
+++ b/drivers/usb/host/ehci-s5p.c
@@ -228,6 +228,8 @@ static int s5p_ehci_suspend(struct device *dev)
  int rc;
 
  rc = ehci_suspend(hcd, do_wakeup);
+ if (rc)
+  return rc;
 
  if (s5p_ehci->otg)
   s5p_ehci->otg->set_host(s5p_ehci->otg, &hcd->self);
-- 
1.7.1